load-gen/src/main/java/io/smartcat/data/loader/worker/AsyncWorker.java

```diff
@@ -0,0 +1,290 @@
+package io.smartcat.data.loader.worker;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import io.smartcat.data.loader.api.AlreadyClosedException;
+import io.smartcat.data.loader.api.Worker;
+import io.smartcat.data.loader.util.LinkedEvictingBlockingQueue;
+
+/**
+ * Asynchronous worker which uses queue and thread pool to schedule work for delegate worker. When queue is full and new
+ * packet is received, an old packet will be dropped. <code>dropFromHead</code> parameter determines whether packet from
+ * head or from tail will e dropped.
+ *
+ * @param <T> Type of data this worker accepts.
+ */
+public class AsyncWorker<T> implements Worker<T>, AutoCloseable {
+
+    private final Consumer<WorkerMeta<T>> workerStatsGetherer;
+    private final LinkedEvictingBlockingQueue<DefaultWorkerMeta> queue;
```

Comment:
Although this is a clean approach (to implement the necessary queue implementation from scratch), I would argue that it's not necessary in this context. Using a LinkedBlockingDequeue instance directly in AsyncWorker implementation and the eviction logic stated within the Worker#accept() method could be a leaner approach.